gobblin-runtime/src/main/java/gobblin/runtime/JobState.java

```diff
@@ -564,8 +564,12 @@ public DatasetState newDatasetState(boolean fullCopy) {
   }
 
   /**
-   * A subclass of {@link JobState} that is used to represent dataset states. This class is currently
-   * identical to {@link JobState} except that the name is more meaningful and less confusing.
+   * A subclass of {@link JobState} that is used to represent dataset states.
+   *
+   * <p>
+   *   A {@code DatasetState} does <em>not</em> contain any properties. Operations such as {@link #getProp(String)}
+   *   and {@link #setProp(String, Object)} are not supported.
+   * </p>
    */
   public static class DatasetState extends JobState {
```

Comment:
Does `DatasetState` need to extend `JobState` anymore? I'm not sure why it does in the first place, a `DatasetState` should need to know things like `jobId` and doesn't really have a `startTime` or an `endTime`.